pyproj/_transformer.pyx

```diff
@@ -30,47 +37,53 @@ cdef class _Transformer:
         if self.projctx is not NULL:
             proj_context_destroy(self.projctx)
 
-    def set_radians_io(self):
-        self.input_radians = proj_angular_input(self.projpj, PJ_FWD)
-        self.output_radians = proj_angular_output(self.projpj, PJ_FWD)
+    def _set_direction(self, direction):
+        try:
+            self.pj_direction = _PJ_DIRECTION_MAP[direction]
+        except KeyError:
+            raise ValueError(
+                "Invalid direction supplied '{}'. "
+                "Only {} are supported."
+                .format(direction, tuple(_PJ_DIRECTION_MAP)))
+
+    def _set_radians_io(self):
+        self.input_radians = proj_angular_input(self.projpj, self.pj_direction)
+        self.output_radians = proj_angular_output(self.projpj, self.pj_direction)
 
     @staticmethod
-    def _init_crs_to_crs(proj_from, proj_to, skip_equivalent=False):
+    def from_crs(crs_from, crs_to, skip_equivalent=False, direction="forward"):
```

Comment:
The `direction` keyword could also be a keyword in the ``(i)transform`` methods instead of here?

Reply:
I thought about that, but for the interest if performance, I put it the way it is. I there weren't many use cases of a repeated transformation where a direction would need to be changed.

Reply:
It reduces the number of calls to `proj_angular_input` and `proj_angular_output`.

Reply:
But, I have not tested how performance impacting this is, only a theory at the moment.

Reply:
Yeah, that's what I wanted ask when I saw your first answer: how does this impact performance? As the argument is not passed to `proj_create_crs_to_crs`, but only when doing the actual in transform in `proj_trans_generic` in the `transform` method.

The advantage of putting it in the transform method is to have it consistent with how PROJ does it (direction is specified when transforming, not when the transform object is created).

Reply:
> It reduces the number of calls to proj_angular_input and proj_angular_output.

There are only two possible directions right (or three with ident)? So we could just store the result for both directions of those functions on the object, and then on transform get the correct one?

Or (thinking aloud here), is this is not always just swapped for inverse? 
As storing those attributes on the Transformer object is still useful I think, but should just always be for the forward version?

Reply:
I prefer to keep as much of the logic internally as simple as possible. Even lookups based on the direction have the potential to be costly with repeated transformations. Many users need the performance. My preference would be the simplest and most performant solution. So, probably need some performance testing for different solutions.

Reply:
Looks like my assumptions were incorrect in this case and the lookup method you provided is a great way to go. It definitely makes the Transformer more flexible and I am glad you brought this up!

Reply:
I believe the current changes resolve this. Thanks again for bringing this up!